### PR TITLE
Refactor (frontend): typed-components

### PR DESCRIPTION
**Pull request by luixo**
Step 1 of [getting complete typescript coverage](https://github.com/near/near-explorer/issues/819)

### CODE REVIEW

File: frontend/src/components/transactions/TransactionsHeader.jsx [hunk 1]
```diff
@@ -1,30 +0,0 @@
-import { Row, Col } from "react-bootstrap";
```
Comment (luixo) [inline frontend/src/components/transactions/TransactionsHeader.jsx:1]: Those files were not used in the project.
Should we create an issue to have a PR check for unused modules?

File: frontend/src/components/utils/Balance.tsx [hunk 2]
```diff
@@ -1,10 +1,19 @@
 /// Copied from near-wallet project:
```
Comment (luixo) [inline frontend/src/components/utils/Balance.tsx:1]: Usage of `BN / string` in this file is kinda messy. We should address it later.

File: frontend/src/components/utils/HeaderNetworkDropdown.tsx [hunk 3]
```diff
@@ -18,13 +24,13 @@ const HeaderDropdownItem = ({ link, title }) => {
...
 
-const HeaderNetworkDropdown = () => (
+const HeaderNetworkDropdown: FC = () => (
   <NetworkConsumer>
     {(context) => (
       <Dropdown>
         <Dropdown.Toggle className="header-network" variant="secondary">
           <div className="network-icon"></div>
-          {context.currentNearNetwork.name}
+          {context.currentNearNetwork?.name}
```
Comment (luixo) [inline frontend/src/components/utils/HeaderNetworkDropdown.tsx:33]: It seems like this file shouldn't be rendered if context is empty.
Though context should either provide sane fallbacks or props should be passed from the outer component.
Another issue to address later.

File: frontend/src/components/utils/MobileHeaderNavDropdown.tsx [hunk 4]
```diff
@@ -59,13 +65,15 @@ const MobileNavItem = ({ link, icon, text }) => {
 
 class MobileNavDropdown extends Component {
   state = { showMenu: false };
+  dropdownWrapper: HTMLDivElement | null = null;
+  dropdownMenu: HTMLDivElement | null = null;
 
-  showMenu = (event) => {
+  showMenu = (event: MouseEvent | React.MouseEvent<HTMLDivElement>) => {
```
Comment (luixo) [inline frontend/src/components/utils/MobileHeaderNavDropdown.tsx:71]: We have mixed input type (react / non-react events).

File: frontend/src/components/utils/Timer.tsx [hunk 5]
```diff
@@ -1,8 +1,19 @@
...
 import Moment from "../../libraries/moment";
 
-class Timer extends PureComponent {
-  constructor(props) {
+interface Props {
+  time?: number;
+}
+
+interface State {
+  time: Date | number;
```
Comment (luixo) [inline frontend/src/components/utils/Timer.tsx:9]: Mixed usage as well, we should make it more strict

File: frontend/src/components/utils/HeaderNetworkDropdown.tsx (hunk 3 above)
Comment (frol) [reply frontend/src/components/utils/HeaderNetworkDropdown.tsx:33]: Good catch

File: frontend/src/components/utils/Search.tsx [hunk 6]
```diff
@@ -79,7 +87,7 @@ class Search extends Component {
   render() {
     return (
       <form
-        onSubmit={this.handleSearch}
+        onSubmit={(e) => this.handleSearch(e)}
```
Comment (frol) [inline frontend/src/components/utils/Search.tsx:90]: Hmm, this way we will have a new function-handler on every render. Why do we need to wrap the handler?

**Review by frol — APPROVED** [review]
I agree with all the risen concerns

File: frontend/src/components/utils/Search.tsx (hunk 6 above)
Comment (luixo) [reply frontend/src/components/utils/Search.tsx]: Yep, probably I tried to figure out things and forgot to clean up :)

**Review by shelegdmitriy — APPROVED** [review]

**Comment by shelegdmitriy** [comment]
@luixo LGTM!

**Pull request merged**